### PR TITLE
Change the dialog layout and styling

### PR DESCRIPTION
**Pull request by MostafaBalata**
I've changed the Dialog according to the new requirements.

Normal Screen
Screen Shot 2019-11-19 at 3 01 17 PM

Responsive
Screen Shot 2019-11-19 at 3 11 58 PM

### CODE REVIEW

File: src/dialog/dialog.tsx [hunk 1]
```diff
@@ -5,11 +5,19 @@ import { onOuterClick } from '@ampel-ui/subscriptions';
 
 const KEY_ESCAPE = 27;
 
+enum DialogLevel {
```
Comment (bedakb) [inline src/dialog/dialog.tsx]: maybe rename to the `DialogType`?

File: src/dialog/dialog.less [hunk 2]
```diff
@@ -12,49 +12,121 @@
     flex-direction: row;
     background: rgba(255, 255, 255, 0.85);
     z-index: 10;
+  
+    &-title {
+      font-weight: normal;
+      font-size: 20px;
+      color: #58575a;
```
Comment (bedakb) [inline src/dialog/dialog.less]: Please use variables.

File: src/dialog/dialog.less [hunk 3]
```diff
@@ -12,49 +12,121 @@
...
+      margin: 0px;
+      -ms-word-break: break-all;
```
Comment (bedakb) [inline src/dialog/dialog.less]: We are using autoprefixer, so no need for this.

File: src/dialog/dialog.less [hunk 4]
```diff
@@ -12,49 +12,121 @@
...
+      word-break: break-all;
```
Comment (bedakb) [inline src/dialog/dialog.less]: there are two same props, which one you want to use ?

File: src/dialog/dialog.less [hunk 5]
```diff
@@ -12,49 +12,121 @@
...
+      &-header {
+        background-color: @white;
+        padding: 0px;
+        color: @gray-coal;
+      }
+  
+      &-body {
+        background-color: @white;
+        font-size: 14px;
+        color: #828182;
```
Comment (bedakb) [inline src/dialog/dialog.less]: :(

File: src/dialog/dialog.tsx [hunk 6]
```diff
@@ -50,23 +58,25 @@ class Dialog extends React.Component<Props, {}> {
...
-                                className="btn btn-secondary"
-                                onClick={this.props.onCancel}
-                            />
-                            <Button
-                                id="success"
-                                text={this.props.btnConfirmText}
-                                className="btn btn-primary"
-                                onClick={this.props.onConfirm}
-                            />
+                        <Button id="cancel" text={'x'} className={'cancel'} onClick={this.props.onCancel} />
```
[the remark this answers is not in the document]
Comment (MostafaBalata) [reply src/dialog/dialog.tsx]: Icon would be better, but Button component doesn't accept icons.

[the remark this answers is not in the document]
Comment (bedakb) [reply src/dialog/dialog.tsx]: The component does not need to accept icon as a prop.
Implement it just like every other icon we are using.

File: src/dialog/dialog.less [hunk 7]
```diff
@@ -1,60 +1,130 @@
...
-            font-size: 12px;
-            padding: 5px 30px;
-        }
-        &-footer {
-            background-color: @white;
-            border: none;
-            padding: 20px 30px 25px;
-            text-align: right;
-        }
+  position: fixed;
```
Comment (bedakb) [inline src/dialog/dialog.less]: Still, indentation is wrong.
Tab == 4 spaces

File: src/dialog/dialog.test.tsx [hunk 8]
```diff
@@ -160,15 +158,15 @@ describe('Dialog', () => {
...
+        expect(onCancel).not.toHaveBeenCalledTimes(1);
     });
 
-    it('should not close the dialog on click inside', () => {
+    it('should render the dialog level', () => {
         const id = 'my-dialog';
         const title = 'My dialog';
         const content = 'Message';
-        const btnCancelText = 'Cancel';
+        const level = DialogType.info;
```
Comment (bedakb) [inline src/dialog/dialog.test.tsx]: It's not a `level` anymore, It's a `type`.

File: src/dialog/dialog.tsx [hunk 9]
```diff
@@ -5,11 +5,19 @@ import { onOuterClick } from '@ampel-ui/subscriptions';
...
+enum DialogType {
+    info = 'info',
+    success = 'success',
+    warning = 'warning',
+}
+
 interface Props {
     id: string;
     title: string;
+    level?: DialogType;
```
Comment (bedakb) [inline src/dialog/dialog.tsx]: `type` ?

File: src/dialog/dialog.tsx [hunk 10]
```diff
@@ -50,23 +58,27 @@ class Dialog extends React.Component<Props, {}> {
...
-                                text={this.props.btnConfirmText}
-                                className="btn btn-primary"
-                                onClick={this.props.onConfirm}
-                            />
+                        <button id="cancel" className={'cancel'} onClick={this.props.onCancel}>
+                            <span className={`icon icon-close`} />
+                        </button>
+                        {this.props.type && (
+                            <div className="dialog-content-icon-wrapper" data-qa={`type-${this.props.type}`}>
+                                <span className={`icon icon-${this.props.type}`} />
```
Comment (BenjaminEckardt) [inline src/dialog/dialog.tsx]: @MostafaBalata I'm a bit late, i know, but when type is not set this will result in `icon-undefined` right?

**Pull request merged**